libcfnet/conn_cache.c

```diff
@@ -105,26 +116,27 @@ AgentConnection *ConnCache_FindIdleMarkBusy(const char *server,
                   "FindIdle: NULL connection in ConnCache_entry!");
 
 
-        if (strcmp(server,  svp->conn->this_server) == 0 &&
-            ConnectionFlagsEqual(&flags, &svp->conn->flags) &&
-            (port == svp->conn->this_port
-             ||
-             (port != NULL && svp->conn->this_port != NULL &&
-              strcmp(port,    svp->conn->this_port) == 0)))
+        if (svp->status == CONNCACHE_STATUS_BUSY)
         {
-            if (svp->status == CONNCACHE_STATUS_BUSY)
-            {
-                Log(LOG_LEVEL_DEBUG, "FindIdle:"
-                    " connection to '%s' seems to be busy.",
-                    server);
-            }
-            else if (svp->status == CONNCACHE_STATUS_OFFLINE)
-            {
-                Log(LOG_LEVEL_DEBUG, "FindIdle:"
-                    " connection to '%s' is marked as offline.",
-                    server);
-            }
-            else if (svp->conn->conn_info->sd >= 0)
+            Log(LOG_LEVEL_DEBUG, "FindIdle:"
+                " connection to '%s' seems to be busy.",
```

Comment:
In `BUSY`, `OFFLINE` and `BROKEN` cases, the connection may not be a connection to `server` as it has not been compared yet with these changes.

Reply:
Good catch, thanks!